{
    "url": "https://api.github.com/repos/spack/spack/issues/1427",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/1427/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/1427/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/1427/events",
    "html_url": "https://github.com/spack/spack/issues/1427",
    "id": 169183456,
    "node_id": "MDU6SXNzdWUxNjkxODM0NTY=",
    "number": 1427,
    "title": "py-meep does not properly find MPI",
    "user": {
        "login": "becker33",
        "id": 13971568,
        "node_id": "MDQ6VXNlcjEzOTcxNTY4",
        "avatar_url": "https://avatars.githubusercontent.com/u/13971568?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/becker33",
        "html_url": "https://github.com/becker33",
        "followers_url": "https://api.github.com/users/becker33/followers",
        "following_url": "https://api.github.com/users/becker33/following{/other_user}",
        "gists_url": "https://api.github.com/users/becker33/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/becker33/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/becker33/subscriptions",
        "organizations_url": "https://api.github.com/users/becker33/orgs",
        "repos_url": "https://api.github.com/users/becker33/repos",
        "events_url": "https://api.github.com/users/becker33/events{/privacy}",
        "received_events_url": "https://api.github.com/users/becker33/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 446619758,
            "node_id": "MDU6TGFiZWw0NDY2MTk3NTg=",
            "url": "https://api.github.com/repos/spack/spack/labels/python",
            "name": "python",
            "color": "c2e0c6",
            "default": false,
            "description": null
        },
        {
            "id": 446779717,
            "node_id": "MDU6TGFiZWw0NDY3Nzk3MTc=",
            "url": "https://api.github.com/repos/spack/spack/labels/rpath",
            "name": "rpath",
            "color": "bfdadc",
            "default": false,
            "description": null
        }
    ],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 4,
    "created_at": "2016-08-03T17:08:39Z",
    "updated_at": "2019-12-09T16:08:52Z",
    "closed_at": "2019-12-09T16:08:51Z",
    "author_association": "MEMBER",
    "active_lock_reason": null,
    "body": "As @adamjstewart wrote in his PR:\n\n\"\"\"\n\nI don't think this actually works. The package is linked to MPI properly:\n\n$ ldd py-meep-1.4.2-ikjdpnikleqeaxvncnag37bhhdnfafoj/lib/python2.7/site-packages/_meep_mpi.so \n...\n    libmpi.so.20 => /blues/gpfs/home/software/spack-0.9.1/opt/spack/linux-centos6-x86_64/gcc-6.1.0/openmpi-2.0.0-tapt7aozvnpq4m7tf7cbsa5zefy4kc6s/lib/libmpi.so.20 (0x00002b0cbd86b000)\n...\n\nbut when I import it:\n\n$ python\nPython 2.7.12 (default, Jul 21 2016, 15:23:03) \n[GCC 6.1.0] on linux2\nType \"help\", \"copyright\", \"credits\" or \"license\" for more information.\n\n> > > import meep_mpi\n> > > Neither libmpi.so.0 nor libmpi.so found. Fatal error.\n> > > Traceback (most recent call last):\n> > >   File \"<stdin>\", line 1, in <module>\n> > >   File \"/soft/spack-0.9.1/opt/spack/linux-centos6-x86_64/gcc-6.1.0/python-2.7.12-hfkksvc2jgtlvkjagr33k6khrnj7yfqt/lib/python2.7/site-packages/meep_mpi.py\", line 5088, in <module>\n> > >     raise e\n> > > OSError: libmpi.so: cannot open shared object file: No such file or directory\n\nit can't find MPI.\n\nBy the way, if I add MPI to my LD_LIBRARY_PATH, py-meep finds it, so I don't think this problem should hold up the PR.\n\n\"\"\"\n\nSPACK packages should not rely on LD_LIBRARY_PATH\n",
    "performed_via_github_app": null
}